haystack/document_stores/opensearch.py

```diff
@@ -178,7 +194,17 @@ def __init__(
         if knn_engine not in {"nmslib", "faiss", "score_script"}:
             raise ValueError(f"knn_engine must be either 'nmslib', 'faiss' or 'score_script' but was {knn_engine}")
 
+        if index_type in ["flat", "hnsw", "ivf", "ivf_pq"]:
+            if index_type in ["ivf", "ivf_pq"] and knn_engine != "faiss":
+                raise DocumentStoreError("Use 'faiss' as knn_engine when using 'ivf' as index_type.")
+            self.index_type = index_type
+        else:
+            raise DocumentStoreError(
+                "Invalid value for index_type in constructor. Choose one of these values: 'flat', 'hnsw', 'ivf', or 'ivf_pq'."
```

Comment:
Nitpicking here but perhaps we can define this list up there before if statement and use it here in the error?